File: Distributed Systems/event_driven_architecture/java/examples/package-tracker-mono-backend/customer-service/src/main/java/com/tracker/customer_service/controller/CreateOrderRequest.java
package com.tracker.customer_service.controller;

import com.tracker.customer_service.entity.Address;
import com.tracker.customer_service.entity.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class CreateOrderRequest {
    private String orderId;
    private UserRef user;
    private OrderStatus status;
    private BigDecimal totalAmount;
    private Address shippingAddress;

    @Data
    public static class UserRef {
        private Long id;
    }
}

File: Distributed Systems/event_driven_architecture/java/examples/package_tracker_frontend/lib/models/create_order_request.dart
import 'address.dart';

class CreateOrderRequest {
  final String? orderId;
  final int userId;
  final String? status;
  final double totalAmount;
  final Address? shippingAddress;

  CreateOrderRequest({
    this.orderId,
    required this.userId,
    this.status,
    required this.totalAmount,
    this.shippingAddress,
  });

  Map<String, dynamic> toJson() => {
    if (orderId != null) 'orderId': orderId,
    'user': {'id': userId},
    if (status != null) 'status': status,
    'totalAmount': totalAmount,
    if (shippingAddress != null) 'shippingAddress': shippingAddress!.toJson(),
  };
}
